core/src/test/java/org/infinispan/distribution/rehash/NonTxPutIfAbsentDuringJoinStressTest.java

```diff
@@ -45,40 +47,35 @@ protected void createCacheManagers() throws Throwable {
    }
 
    private ConfigurationBuilder getConfigurationBuilder() {
-      ConfigurationBuilder c = new ConfigurationBuilder();
-      c.clustering().cacheMode(CacheMode.DIST_SYNC);
-      c.transaction().transactionMode(TransactionMode.NON_TRANSACTIONAL);
-      return c;
+      return getDefaultClusteredCacheConfig(cacheMode, false);
    }
 
-   @Test(groups = "unstable", description = "See ISPN-3918")
+// The test is unstable only in dist/repl cache, scattered cache is fine
```

Comment:
The test is correct (up to our expectations how a cache should behave), the problem is in the code - see ISPN-3918. And TBH I have no idea how to fix that besides always going to primary owner. Scattered cache does exaclty that.
Unstable is effectively disabling the test anyway, and I am keeping the previous part commented for that reason.